snapcraft/internal/errors.py

```diff
@@ -166,6 +166,14 @@ class MissingGadgetError(SnapcraftError):
         'https://github.com/snapcore/snapd/wiki/Gadget-snap')
 
 
+class PluginOutdatedError(SnapcraftError):
+
+    fmt = 'This plugin is outdated: {message}'
+
+    def __init__(self, message):
```

Comment:
What kind of errors are you referring to here? Maybe we can add some validation to the base class?